.github/workflows/earthly-build.yaml

```diff
@@ -42,10 +42,17 @@ jobs:
         run: |
           earthly --ci --output +sources
           earthly --ci +image
-      - name: Push spaceros image
+      - name: Push tagged spaceros images to Dockerhub
         env:
           DOCKER_HUB_TOKEN: ${{ secrets.DOCKER_HUB_RW_TOKEN }}
         if: ${{ github.ref_type == 'tag' }}
         run: |
           echo $DOCKER_HUB_TOKEN | docker login --username osrfbot --password-stdin
           earthly --ci --push +image --tag=${{ github.ref_name }}
+      - name: Push the main spaceros image to GitHub
+        # TODO: Revert this post-testing
+        if: ${{ github.ref_name == '194/merge' }}
+        run: |
+          echo "${{ secrets.GITHUB_TOKEN }}" | docker login ghcr.io -u ${{ github.actor }} --password-stdin
+          docker tag osrf/space-ros:latest ghcr.io/space-ros/space-ros:main
+          docker push ghcr.io/space-ros/space-ros:push-main-image
```

Comment:
I think the image will not be generated locally unless it is Earthly command with a proper push, not a docker one. I had this issue with my own attempt to push. workaround here would be to change image name in Earthfile and push...

Reply:
Ah gosh yes, I am learning the hard way too. I was hoping to do this minimally but I'll play with it...